src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -102,16 +117,59 @@ public void ValidateConfig()
             // Connection string can't be null or empty
             if (string.IsNullOrWhiteSpace(dataSource.ConnectionString))
             {
-                throw new DataApiBuilderException(
-                    message: DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE,
-                    statusCode: HttpStatusCode.ServiceUnavailable,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                HandleOrRecordException(
+                    new DataApiBuilderException(
+                        message: DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE,
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization)
+                );
             }
         }
 
         ValidateDatabaseType(runtimeConfig, fileSystem, logger);
     }
 
+    public async void ValidateConfig(string configFilePath)
+    {
+        _isValidateOnly = true;
```

Comment:
Keeping a static member `_isValidateOnly` and modifying it via Test, here doesnt seem intuitive. You can create separate object for RuntimeConfigValidator in CLI with this property set to true when its the validate command. For all other cases, you can keep the default value of this property to be false. 

Static members should be defined to share state between different objects - as a class wide property not for convenience of use.